worker/solution.go

```diff
@@ -0,0 +1,128 @@
+package worker
+
+import (
+	"fmt"
+	"os"
+	"errors"
+	"gorm.io/gorm"
+	"github.com/mini-maxit/worker/models"
+)
+
+
+// ErrInvalidFileCount is returned when the solution directory contains more than one solution file
+var ErrInvalidFileCount = errors.New("expected exactly one file in the solution directory")
+
+// ErrFileNotFound is returned when a file is not found
+var ErrFileNotFound = errors.New("file not found")
+
+// GetDataForSolutionRunner retrieves the data needed to run the solution
+func getDataForSolutionRunner(db *gorm.DB, task_id, user_id, user_solution_id, input_output_id int) (models.Task, error) {
+	var task models.Task
+	var dirPath string
+
+	// Fetch data with GORM
+	err := db.Raw(`
+		SELECT dir_path
+		FROM tasks
+		WHERE id = ?
+	`, task_id).Scan(&dirPath).Error
+	if err != nil {
+		return models.Task{}, err
+	}
+
+	err = db.Raw(`
+		SELECT compiler_type
+		FROM user_solutions
+		WHERE id = ?
+	`, user_solution_id).Scan(&task.CompilerType).Error
+	if err != nil {
+		return models.Task{}, err
+	}
+
+	var ioData models.InputOutputData
+
+	err = db.Raw(`
+		SELECT time_limit, memory_limit, input_file_path, output_file_path
+		FROM inputs_outputs
+		WHERE id = ?
+	`, input_output_id).Scan(&ioData).Error
+	if err != nil {
+		return models.Task{}, err
+	}
+
+	task.TimeLimit = ioData.TimeLimit
+	task.MemoryLimit = ioData.MemoryLimit
+	task.Stdin = ioData.InputFilePath
+	task.ExpectedStdout = ioData.OutputFilePath
+	if err != nil {
+		return models.Task{}, err
+	}
+
+	err = getFiles(dirPath, user_id, user_solution_id, task.Stdin, task.ExpectedStdout, &task)
+	if err != nil {
+		return models.Task{}, err
+	}
+
+	return task, nil
+}
+
+// GetFiles retrieves the solution, input, and output files from the file system
+func getFiles(dir_path string, user_id, user_solution_id int, input_path, output_path string, task *models.Task) error {
```

Comment:
Function comment says that you retrieve solution input and output files, however you retrieve only one pair of input/output. Remember that the solution can have multipul inputs/outputs